core/compilePatterns.ml

```diff
@@ -79,12 +79,12 @@ let rec desugar_pattern : Ir.scope -> Sugartypes.pattern -> pattern * raw_env =
     let pp = desugar_pattern scope in
     let empty = (NEnv.empty, TEnv.empty, Types.make_empty_open_row (`Any, `Any)) in
     let (++) (nenv, tenv, _) (nenv', tenv', eff') = (NEnv.extend nenv nenv', TEnv.extend tenv tenv', eff') in
-    let fresh_binder (nenv, tenv, eff) =
-      function
-        | (name, Some t, _) ->
-            let xb, x = Var.fresh_var (t, name, scope) in
-              xb, (NEnv.bind nenv (name, x), TEnv.bind tenv (x, t), eff)
-        | _ -> assert false
+    let fresh_binder (nenv, tenv, eff) bndr =
+      assert (Sugartypes.bndrHasTy bndr);
+      let name = Sugartypes.bndrName bndr in
```

Comment:
Please rename `Sugartypes.bndrName` to `Sugartypes.name_of_binder` or consider introducing a `Binder` module with a `to_name` function within `Sugartypes`.